scapy/arch/windows/__init__.py

```diff
@@ -67,16 +69,58 @@ def _encapsulate_admin(cmd):
     # To get admin access, we start a new powershell instance with admin rights, which will execute the command
     return "Start-Process PowerShell -windowstyle hidden -Wait -Verb RunAs -ArgumentList '-command &{%s}'" % cmd
 
+class _PowershellManager(Thread):
+    """Used only when scapy boots. Instance used to send multiple
+    commands on the same Powershell process. Skips the powershell
+    booting time to speed up scapy's boot"""
+    def __init__(self):
+        self.process = sp.Popen([conf.prog.powershell, "-NoLogo", "-NonInteractive", "-Command", "-"], # Start & redirect input
+                         stdout=sp.PIPE,
+                         stdin=sp.PIPE,
+                         stderr=sp.STDOUT,
+                         universal_newlines=True)
+        self.buffer = []
+        self.running = True
+        Thread.__init__(self)
+        self.start()
+
+    def run(self):
+        while self.running:
+            self.buffer.append(self.process.stdout.readline())
+
+    def query(self, cmd):
+        self.buffer = []
```

Comment:
self.buffer was initialized in __init__ already and it's never destroyed, so there's no need to reinitialize it.

Reply:
Sorry but it's definatly needed, as we execute query multiple times, and each time we get a new buffer to store the result of the command